src/org/jitsi/impl/neomedia/MediaStreamImpl.java

```diff
@@ -3635,6 +3636,89 @@ public void injectPacket(RawPacket pkt, boolean data, TransformEngine after)
     }
 
     /**
+    * {@inheritDoc}
+    */
+    public boolean isKeyFrame(byte[] buf, int off, int len)
+    {
+        if (buf == null || buf.length < off + len)
+        {
+            return false;
+        }
+
+        Byte redPT = null, vp8PT = null, h264PT = null;
+        for (Map.Entry<Byte, MediaFormat> entry :
+            getDynamicRTPPayloadTypes().entrySet())
+        {
+            String encoding = entry.getValue().getEncoding();
+            Byte payloadType = entry.getKey();
+
+            if (Constants.RED.equalsIgnoreCase(encoding))
+            {
+                redPT = payloadType;
+            }
+            else if (Constants.VP8.equalsIgnoreCase(encoding))
+            {
+                vp8PT = payloadType;
+            }
+            else if (Constants.H264.equalsIgnoreCase(encoding))
+            {
+                h264PT = payloadType;
+            }
+        }
+
+        int payloadOff, payloadLen;
+        byte payloadType;
+
+        try {
+          // XXX this will not work correctly when RTX gets enabled!
+            if (redPT != null
+                && redPT == RawPacket.getPayloadType(buf, off, len))
+            {
+                REDBlock block = REDBlockIterator
+                    .getPrimaryBlock(buf,
+                        RawPacket.getPayloadOffset(buf, off, len),
+                        RawPacket.getPayloadLength(buf, off, len));
+
+                payloadOff = block.getOffset();
+                payloadLen = block.getLength();
+                payloadType = block.getPayloadType();
+            }
+            else
+            {
+                payloadOff = RawPacket.getPayloadOffset(buf, off, len);
+                payloadLen = RawPacket.getPayloadLength(buf, off, len);
+                payloadType
+                    = RawPacket.getPayloadType(buf, off, len);
+            }
+        }
+        catch (ArrayIndexOutOfBoundsException e)
+        {
+            // While ideally we should check the bounds everywhere and not
+            // attempt to access the packet's buffer at invalid indexes, there
+            // are too many places where it could inadvertently happen. It's
+            // safer to return a default value of 'false' from this utility
+            // method than to risk killing a thread which doesn't expect this.
+            logger.warn("Caught an exception while checking for keyframe:", e);
+            return false;
+        }
+
+        if (vp8PT != null && payloadType == vp8PT)
```

Comment:
I think we should probably move this if block inside the try-catch. I hate it that we have to catch array boundaries errors like this, we have done a sloppy job, but that's what we have right now.